File: fixtures/scrapy_output/multiline_post_with_data.py
from scrapy import Request, Spider


class MySpider(Spider):
    name = 'myspider'

    def start_requests(self):
        headers = {
            'Origin': 'http://fiddle.jshell.net',
        }
        body = {
            'msg1': 'value1',
            'msg2': 'value2',
        }
        body = '&'.join(key + '=' + value for key, value in body.items())
        yield Request(
            'http://fiddle.jshell.net/echo/html/',
            callback=self.parse,
            method='POST',
            headers=headers,
            body=body,
        )

    def parse(self, response):
        pass

File: test/fixtures/node-superagent/multiline_post_with_data.js
import request from 'superagent';

request
  .get('http://localhost:28139/echo/html/')
  .set('Origin', 'http://fiddle.jshell.net')
  .type('form')
  .send({
    'msg1': 'value1',
    'msg2': 'value2'
  })
  .then(res => {
     console.log(res.body);
  });
